core/GlobalState.cc

```diff
@@ -2027,22 +2028,46 @@ void GlobalState::sanityCheck() const {
             "name table and hash name table sizes out of sync names.capacity={} namesByHash.capacity={}",
             namesUsedTotal(), namesByHash.capacity());
     ENFORCE(namesByHash.size() == namesByHash.capacity(), "hash name table not at full capacity");
+}
 
-    {
-        Timer timeit(tracer(), "GlobalState::sanityCheck (names)");
-        for (uint32_t i = 0; i < utf8Names.size(); i++) {
-            NameRef(*this, NameKind::UTF8, i).sanityCheck(*this);
-        }
+void GlobalState::sanityCheckNames() const {
+    if (!debug_mode) {
+        return;
+    }
+    if (fuzz_mode) {
+        // it's very slow to check this and it didn't find bugs
+        return;
+    }
 
-        for (uint32_t i = 0; i < constantNames.size(); i++) {
-            NameRef(*this, NameKind::CONSTANT, i).sanityCheck(*this);
-        }
+    Timer timeit(tracer(), "GlobalState::sanityCheck (names)");
 
-        for (uint32_t i = 0; i < uniqueNames.size(); i++) {
-            NameRef(*this, NameKind::UNIQUE, i).sanityCheck(*this);
-        }
+    for (uint32_t i = 0; i < utf8Names.size(); i++) {
+        NameRef(*this, NameKind::UTF8, i).sanityCheck(*this);
     }
 
+    for (uint32_t i = 0; i < constantNames.size(); i++) {
+        NameRef(*this, NameKind::CONSTANT, i).sanityCheck(*this);
+    }
+
+    for (uint32_t i = 0; i < uniqueNames.size(); i++) {
+        NameRef(*this, NameKind::UNIQUE, i).sanityCheck(*this);
+    }
+}
+
+void GlobalState::sanityCheck() const {
+    if (!debug_mode) {
+        return;
+    }
+    if (fuzz_mode) {
+        // it's very slow to check this and it didn't find bugs
+        return;
+    }
+
+    Timer timeit(tracer(), "GlobalState::sanityCheck");
+
+    sanityCheckTableSizes();
+    sanityCheckNames();
+
     {
         Timer timeit(tracer(), "GlobalState::sanityCheck (symbols)");
```

Comment:
Any reason not to split this off into `sanityCheckSymbols`? Do we always want to also do `sanityCheckTableSizes` and `sanityCheckNames` when this is done?

Reply:
I'm not quite sure what you're asking.

There were some places where I **only** wanted to sanity check the table sizes, or sanity check the names, so I factored out helpers for these.

There were no places where I **only** wanted to sanity check the symbols, so I did not factor them out of the top-level sanity check function.

Reply:
Ah, that answers exactly what I was asking, thanks.